modules/common/networkattachment/networkattachment.go

```diff
@@ -97,18 +97,20 @@ func GetNetworkStatusFromAnnotation(annotations map[string]string) ([]networkv1.
 
 // VerifyNetworkStatusFromAnnotation - verifies if NetworkStatus annotation for the pods of a deployment,
 // pods identified via the service label selector, matches the passed in network attachments and the number of
-// per network IPs the ready count of the deployment. Return true if count matches with the list of IPs per network.
+// per network IPs the ready count of the deployment. Return true if count matches with the list of IPs per network,
+// no networkAttachments provided or expectedCount == 0
 func VerifyNetworkStatusFromAnnotation(
 	ctx context.Context,
 	helper *helper.Helper,
 	networkAttachments []string,
 	serviceLabels map[string]string,
 	readyCount int32,
+	expectedCount int32,
 ) (bool, map[string][]string, error) {
 
 	networkReady := true
 	networkAttachmentStatus := map[string][]string{}
-	if len(networkAttachments) > 0 {
+	if len(networkAttachments) > 0 && expectedCount > 0 {
```

Comment:
would it be simpler not to  call VerifyNetworkStatusFromAnnotation() from the operator if Replicas is 0?

Reply:
that's an alternative, can set the `networkReady` in the caller to true if Replicas is 0. doing it in here is just a bit more generic?

Reply:
we just need to get an agreement on one of the other way doing it.

Reply:
currently `expectedCount` looks like a boolean flag that turns off the whole logic of `VerifyNetworkStatusFromAnnotation`. So if I refactor this to be a boolean flag 

```golang
func VerifyNetworkStatusFromAnnotation(
	ctx context.Context,
	helper *helper.Helper,
	networkAttachments []string,
	serviceLabels map[string]string,
	readyCount int32,
	skipVerify bool,
) (bool, map[string][]string, error) {

    if skipVerify {
        return true
    }
```
then I think it shows more clearly why I think this does not belong here.